lib/plug/csrf_protection.ex

```diff
@@ -101,19 +101,39 @@ defmodule Plug.CSRFProtection do
 
   @behaviour Plug
 
-  def init(opts), do: opts
+  
+  def init(opts) do
+    verity_opts opts 
+  end
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verity_opts opts
     csrf_token = get_session(conn, "_csrf_token")
     Process.put(:plug_csrf_token, csrf_token)
-
     if not verified_request?(conn, csrf_token) do
-      raise InvalidCSRFTokenError
+      if opts[:with] == :exception do
+        raise InvalidCSRFTokenError
+      else 
+        conn = clear_session conn
+      end
     end
 
     register_before_send(conn, &ensure_same_origin_and_csrf_token!(&1, csrf_token))
   end
 
+  
+  ## verifies if options passes are valid
+  ## :with - should be one of :exception or :nil_session, defaults to :exception     
+  defp verity_opts opts do
```

Comment:
Looks like you also misspelled `verity_opts` here and everywhere else it's called :)